### PR TITLE
Ensure ResourceSynced condition set

### PR DESCRIPTION
**Pull request by jaypipes**
There really isn't an "active" or "available" status for Role resources
in IAM. Basically, when the role is successfully created or updated,
it's in sync with whatever the desired state of the role was. So, this
patch adds simple setters after successful create and update code paths
that ensures ResourceSynced condition of True is set on the Role CRs.

Signed-off-by: Jay Pipes <jaypipes@gmail.com>

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by vijtrip2 — COMMENTED** [review]
/approve

File: test/e2e/condition.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
```
Comment (RedbackThomson) [inline test/e2e/condition.py:1]: Can all of this file go into `acktest` instead? :)

**Review by RedbackThomson — CHANGES_REQUESTED** [review]
This looks like prime re-usable code to me 😄

File: test/e2e/condition.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Utility functions to help processing Kubernetes resource conditions"""
+
+# TODO(jaypipes): Move these functions to acktest library. The reason these are
+# here is because the existing k8s.assert_condition_state_message doesn't
+# actually assert anything. It returns true or false and logs messages.
```
Comment (jaypipes) [inline test/e2e/condition.py:18]: @RedbackThomson see this comment? :)

Comment (RedbackThomson) [reply test/e2e/condition.py:18]: Why is this a blocker for these methods to be moved into acktest though?

Comment (jaypipes) [reply test/e2e/condition.py:18]: it's not a blocker to move into acktest.

File: test/e2e/condition.py (hunk 1 above)
Comment (jaypipes) [reply test/e2e/condition.py:1]: https://github.com/aws-controllers-k8s/test-infra/pull/166

File: test/e2e/condition.py (hunk 2 above)
Comment (jaypipes) [reply test/e2e/condition.py:18]: https://github.com/aws-controllers-k8s/test-infra/pull/166

Comment (RedbackThomson) [reply test/e2e/condition.py:18]: Thanks! Merged!

**Comment by RedbackThomson** [comment]
Looking good. If tests pass then I'm happy.
/lgtm

**Pull request merged**